src/CAN/can_mapping.c

```diff
@@ -47,7 +49,30 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
                         break;
                 }
         }
-        return (float)raw_value;
+
+        /* convert type */
+        float extracted_value = 0;
+        switch (mapping->type) {
+        	case CANMappingType_IEEE754:
+        		extracted_value = *((float*)&raw_value);
+        		break;
+        	case CANMappingType_signed:
+        		if (mapping->length <= 8) {
+        			extracted_value = *((int8_t*)&raw_value);
```

Comment:
I had made the initial request to Brent on this. Ideally you should strive to support U1 to U32's and S2 to S32's on all signals, to stay bit atomic. On OEM cans' there are items like S7's and S9's that would not completely work here.

Currently before this commit there was no int8_t support of variables and this is a step forward. Perhaps a fully agree'd upon functional requirement and then Test case would be the best place to start, to allow the code to follow